publications/tasks.py

```diff
@@ -83,29 +84,34 @@ def parse_oai_xml_and_save_publications(content, event):
         return
     
     for record in records:
-        # Initialize defaults so variables are always defined.
-        period_start, period_end, geom_object = [], [], None
-        
         try:
             def get_text(tag_name):
                 nodes = record.getElementsByTagName(tag_name)
                 return nodes[0].firstChild.nodeValue.strip() if nodes and nodes[0].firstChild else None
-            
-            identifier_value = get_text("dc:identifier")
+
+            # collect all dc:identifier values
+            id_nodes = record.getElementsByTagName("dc:identifier")
+            identifiers = [
+                n.firstChild.nodeValue.strip()
+                for n in id_nodes
+                if n.firstChild and n.firstChild.nodeValue
+            ]
+            http_urls = [u for u in identifiers if u.lower().startswith("http")]
+            view_urls = [u for u in http_urls if "/view/" in u]
+            identifier_value = (view_urls or http_urls or [None])[0]
+
             title_value = get_text("dc:title")
             abstract_text = get_text("dc:description")
             journal_value = get_text("dc:publisher")
             date_value = get_text("dc:date")
+
             doi_text = None
-            doi_nodes = record.getElementsByTagName("dc:identifier")
-            for node in doi_nodes:
-                if node.firstChild and node.firstChild.nodeValue:
-                    candidate = node.firstChild.nodeValue.strip()
-                    match = DOI_REGEX.search(candidate)
-                    if match:
-                        doi_text = match.group(0)
-                        break
-            
+            for ident in identifiers:
+                match = DOI_REGEX.search(ident)
+                if match:
+                    doi_text = match.group(0)
+                    break
+
             # Duplicate checking.
```

Comment:
The duplicate checking is important here, but I don't like the way that you save all existing DOIs in a local variable. This is a potentially costly operation to repeat every time we add a record.

What can be optimised, however, is the database query (cf. for later #141). Therefore, instead of checking if the DOI is in the list, please query the database with the DOI and then assume the record does not exist when the result set is empty.

You should be able to combine the query with the URL.